lib/search/document.js

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module search/document
+ */
+
+'use strict';
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+function Document(index, id) {
+  this.search_ = index.search_;
+  this.index_ = index;
+  this.id = id;
+}
+
+Document.prototype.delete = function(callback) {
+  callback = callback || util.noop;
+  this.makeReq_('DELETE', '', null, null, callback);
+};
+
+Document.prototype.getMetadata = function(callback) {
+  var self = this;
+
+  callback = callback || util.noop;
+
+  this.makeReq_('GET', '/', null, null, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    if (util.is(resp.fields, 'object')) {
+      self.fields = resp.fields;
+    }
+
+    if (util.is(resp.rank, 'number')) {
+      self.rank = resp.rank;
+    }
+
+    callback(null, self, resp);
+  });
+};
+
+Document.prototype.makeReq_ = function(method, path, query, body, callback) {
+  path = '/documents/' + this.id + path;
```

Comment:
Yeah, we are re-using the `makeReq_` from the index, so the index components are already in place.